packages/docmanager-extension/src/index.tsx

```diff
@@ -857,8 +866,20 @@ function addCommands(
           if (saveInProgress.has(context)) {
             return;
           }
-
-          if (context.model.readOnly) {
+          if (!context.contentsModel?.writable) {
+            let type;
+            if (args._luminoEvent) {
+              type = (args._luminoEvent as ReadonlyPartialJSONObject).type;
+            }
+            if (type === 'keybinding') {
+              return Notification.warning(
+                trans.__(
+                  `%1 is permissioned as read-only. Use "save as..." instead.`,
```

Comment:
The other notification says: `%1 is permissioned as readonly. Use "save as..." instead.` (readonly vs read-only). Would it make sense to move creation of the warning to a function to ensure that only one version of the string is used?